packages/ember-application/lib/system/application.js

```diff
@@ -1097,4 +1011,67 @@ function resolverFor(namespace) {
   return resolve;
 }
 
+function registerLibraries() {
+  if (!librariesRegistered) {
+    librariesRegistered = true;
+
+    if (environment.hasDOM) {
+      Ember.libraries.registerCoreLibrary('jQuery', jQuery().jquery);
+    }
+  }
+}
+
+function logLibraryVersions() {
+  if (Ember.LOG_VERSION) {
+    // we only need to see this once per Application#init
+    Ember.LOG_VERSION = false;
+    var libs = Ember.libraries._registry;
+
+    var nameLengths = EnumerableUtils.map(libs, function(item) {
+      return get(item, 'name.length');
+    });
+
+    var maxNameLength = Math.max.apply(this, nameLengths);
+
+    Ember.debug('-------------------------------');
+    for (var i = 0, l = libs.length; i < l; i++) {
+      var lib = libs[i];
+      var spaces = new Array(maxNameLength - lib.name.length + 1).join(' ');
+      Ember.debug([lib.name, spaces, ' : ', lib.version].join(''));
+    }
+    Ember.debug('-------------------------------');
+  }
+}
+
+var ApplicationInstance = Ember.Object.extend({
+  container: null,
+  customEvents: null,
+  rootElement: null,
+
+  startRouting: function(isModuleBasedResolver) {
+    var router = this.container.lookup('router:main');
+    if (!router) { return; }
+
+    router.startRouting(isModuleBasedResolver);
+  },
+
+  handleURL: function(url) {
+    var router = this.container.lookup('router:main');
+
+    return router.handleURL(url);
+  },
+
+  setupEventDispatcher: function() {
+    var dispatcher = this.container.lookup('event_dispatcher:main');
+
+    dispatcher.setup(this.customEvents, this.rootElement);
+
+    return dispatcher;
+  },
+
+  willDestroy: function() {
+    run(this.container, 'destroy');
```

Comment:
This is just preserving semantics you introduced in 9c83112d9698823e4b9f540891a1f083d17d9503. We specifically were trying to hew to existing behavior as much as possible without doing a more in-depth refactoring.